lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/Versions.java

```diff
@@ -32,10 +32,10 @@ private Versions() {}
      * chooses not to maintain compatibility with older SPI versions, as happens for plugins maintained together with
      * the Trino project.
      */
-    public static void checkSpiVersion(ConnectorContext context, ConnectorFactory connectorFactory)
+    public static void checkSpiVersion(ConnectorContext context, String connectorName, Class<? extends Connector> connectorClass)
```

Comment:
Changes in `trino-plugin-toolkit` require anyone who uses the toolkit to recompile / fix the compilation errors before they can uptake a new version (i.e., every release). Is this necessary?

Reply:
This library is not considered as SPI in regards of backwards compatibility. It is just utility that plugins may or may not use. Typically we change this library to match our needs, backwards compatibility is a second priority here.

Reply:
Agreed, backwards compatibility is not an issue for plugin toolkit.